app/fetchfile.py

```diff
@@ -0,0 +1,42 @@
+import os
+
+import docx
+import requests
+from flask import send_file, send_from_directory
+
+
+class Fetchfile(object):
+
+    def __init__(self, app=None):
+        self.app = app
+        self._linker_url = None
+        self._base_dir = None
+
+        if app is not None:
+            self.init_app(app)
+
+    def init_app(self, app):
+        self._base_dir = self.app.config['BASE_DIR']
+        self._linker_url = self.app.config['LINKER_URL']
+        app.fetchfile = self
+
+    def fetch_doc(self, uid, outdir):
+        if not self.file_is_doc(uid):
+            return Exception
```

Comment:
In python we are [raising-exceptions](https://docs.python.org/3/tutorial/errors.html#raising-exceptions). Not returning them.
Also add some nice message to identify the problem.